metrics-operator/controllers/common/providers/dynatrace/dynatrace_test.go

```diff
@@ -648,88 +751,37 @@ func TestEvaluateQueryForStep_RefNotExistingKey(t *testing.T) {
 			secretKey: []byte(secretValue),
 		},
 	}
-	kdp, obj := setupTestForTimerangeWithStep(apiToken)
+	kdp := setupTest(apiToken)
 
-	missingKey := "key_not_found"
-	p := metricsapi.KeptnMetricsProvider{
-		Spec: metricsapi.KeptnMetricsProviderSpec{
-			SecretKeyRef: v1.SecretKeySelector{
-				LocalObjectReference: v1.LocalObjectReference{
-					Name: secretName,
-				},
-				Key: missingKey,
+	keptnMetric := metricsapi.KeptnMetric{
+		Spec: metricsapi.KeptnMetricSpec{
+			Query: "my-query",
+			Range: &metricsapi.RangeSpec{
+				Interval:    "5m",
+				Step:        "1m",
+				Aggregation: "max",
 			},
-			TargetServer: svr.URL,
-		},
-	}
-	_, _, e := kdp.EvaluateQueryForStep(context.TODO(), obj, p)
-	require.NotNil(t, e)
-	require.True(t, strings.Contains(e.Error(), "invalid key "+missingKey))
-}
-
-func TestEvaluateQuery_HappyPathForTimerangeWithStep(t *testing.T) {
-	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		_, err := w.Write([]byte(dtpayload))
-		p := r.URL.Query().Get("from")
-		require.NotNil(t, p)
-		require.Nil(t, err)
-	}))
-	defer svr.Close()
-	secretName, secretKey, secretValue := "secretName", "secretKey", "secretValue"
-	apiToken := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      secretName,
-			Namespace: "",
-		},
-		Data: map[string][]byte{
-			secretKey: []byte(secretValue),
 		},
 	}
-	kdp, obj := setupTestForTimerangeWithStep(apiToken)
 
+	missingKey := "key_not_found"
 	p := metricsapi.KeptnMetricsProvider{
 		Spec: metricsapi.KeptnMetricsProviderSpec{
 			SecretKeyRef: v1.SecretKeySelector{
 				LocalObjectReference: v1.LocalObjectReference{
 					Name: secretName,
 				},
-				Key: secretKey,
+				Key: missingKey,
 			},
 			TargetServer: svr.URL,
 		},
 	}
-	r, raw, e := kdp.EvaluateQueryForStep(context.TODO(), obj, p)
-	require.Nil(t, e)
-	require.Equal(t, []byte(dtpayload), raw)
-	require.Equal(t, []string{"50.000000", "50.000000", "50.000000", "50.000000", "50.000000", "50.000000", "50.000000", "50.000000"}, r)
-}
-
-func setupTest(objs ...client.Object) (KeptnDynatraceProvider, []metricsapi.KeptnMetric) {
-
-	fakeClient := fake.NewClient(objs...)
-
-	kdp := KeptnDynatraceProvider{
-		HttpClient: http.Client{},
-		Log:        ctrl.Log.WithName("testytest"),
-		K8sClient:  fakeClient,
-	}
-	objects := []metricsapi.KeptnMetric{
-		{
-			Spec: metricsapi.KeptnMetricSpec{
-				Query: "my-query",
-			},
-		},
-		{
-			Spec: metricsapi.KeptnMetricSpec{
-				Query: "my-query",
-				Range: &metricsapi.RangeSpec{Interval: "5m"},
-			},
-		},
-	}
-	return kdp, objects
+	_, _, e := kdp.EvaluateQueryForStep(context.TODO(), keptnMetric, p)
+	require.NotNil(t, e)
+	require.True(t, strings.Contains(e.Error(), "invalid key "+missingKey))
 }
 
-func setupTestForTimerange(objs ...client.Object) (KeptnDynatraceProvider, metricsapi.KeptnMetric) {
+func setupTest(objs ...client.Object) KeptnDynatraceProvider {
```

Comment:
I reduced the `setupTest...` functions to one `setupTest` function that only takes care of setting up the environment for the provider component. The creation of the KeptnMetrics object has been removed deliberately, to make the individual unit tests more readable as the arguments (i.e. the KeptnMetric objects) are initialized directly in the test functions - this makes it easier to comprehend what the test cases should verify